src/Xenial.Tasty.Cli/Program.cs

```diff
@@ -0,0 +1,236 @@
+                            {
+                                var commands = await waitForCommands(server);
+                                await Task.Run(async () => await WaitForInput(commands, server), cancellationToken);
+                            }, cancellationToken);
+
+                            await Task.WhenAll(remoteTask, uiTask);
+                        }
+                        catch (TaskCanceledException)
+                        {
+                            return 0;
+                        }
+                        catch (SimpleExec.NonZeroExitCodeException e)
+                        {
+                            return e.ExitCode;
+                        }
+                    }
+                }
+                return 0;
+            }
+            catch (TaskCanceledException)
+            {
+                return 1;
+            }
+        }
+
+        static Task WaitForInput(IList<SerializableTastyCommand> commands, TastyServer tastyServer)
+            => Promise((resolve) =>
+            {
+                Func<Task> cancelKeyPress = () => Promise((resolve, reject) =>
+               {
+                   Console.CancelKeyPress += (_, e) =>
+                   {
+                       if (e.Cancel)
+                       {
+                           Console.WriteLine("Cancelling execution...");
+                           reject();
+                       }
+                   };
+               });
+
+                Func<Task> endTestPipelineSignaled = () => Promise((resolve, reject) =>
+                {
+                    tastyServer.EndTestPipelineSignaled = () =>
+                    {
+                        Console.WriteLine("Cancelling execution...");
+                        reject();
+                    };
+                });
+
+                Func<Task<ConsoleKeyInfo>> readConsoleKey = () => Promise<ConsoleKeyInfo>(resolve =>
+                {
+                    _ = Task.Run(() =>
+                    {
+                        var info = Console.ReadKey(true);
+                        resolve(info);
+                    });
+                });
+
+                Action writeCommands = () =>
+                {
+                    Console.WriteLine("Interactive Mode");
+                    Console.WriteLine("================");
+
+                    foreach (var c in commands)
+                    {
+                        Console.WriteLine($"{c.Name} - {c.Description}" + (c.IsDefault ? " (default)" : string.Empty));
+                    }
+
+                    Console.WriteLine("c - Cancel");
+                    Console.WriteLine("================");
+                };
+
+                Func<ConsoleKeyInfo, SerializableTastyCommand?> findCommand = (info) =>
+                {
+                    var key = info.Key.ToString().ToLower();
+
+                    var command = info.Key == ConsoleKey.Enter
+                        ? commands.FirstOrDefault(c => c.IsDefault)
```

Comment:
I think we should guard for more than one default commands. Latest default should win and should be part of the API contract. So we need to guard that a the client registration phase

Reply:
Done